routes.js

```diff
@@ -1,11 +1,17 @@
 const express = require('express');
+const jwt = require('jsonwebtoken');
+const { privateKey } = require('./config/apiCredentials');
 
 const router = express.Router();
 
 router.use('/login', express.static('frontend/login.html'));
 
 router.post('/authenticate', (request, response) => {
-	response.send(request.body);
+	const { username , password } = request.body;
+	const token = jwt.sign({username, password}, privateKey );
+
+	response.header('Authorization', 'bearer ' +token);
```

Comment:
Instead of putting the token in the Authorization header, what we can do is set it in the cookie, which will mean the browser will keep track of it without needing us to manually do anything, and the browser will also send it along with any requests it makes in the future.

So something like...

```
response.cookie("access_token", token, { httpOnly: true });
```

The httpOnly bit means that the cookie can't be accessed by any JavaScript on the page, which means JavaScript can't try and steal the user's credentials.

Reply:
In the future, we'd check if the cookie was there just by looking at

```
request.cookies.access_token
```

Reply:
Done